### ISSUE TITLE
xAxis rotate unhides the axis labels only on small window size

### ISSUE THREAD

**Issue by xerxes3117** [apache/echarts#9723]
### Version & Environment [Version & Environment]

- ECharts version [ECharts version]: 4.0
- Browser version [browser type and version]: chrome
- OS Version [operating system type and version]: windows 10

### One-line summary [Question Brief]
I'm trying to rotate the labels to unhide the x axis labels. I want to rotate the labels based on some conditions so I do this after initially setting the options (using setTimeout). The labels rotate works but the issue is that labels are not unhidden. Additionally, after some tinkering I found a weird behavior. This issue is not present if the container window is small.

### Expected behaviour [expected result]
Axis labels should be unhidden in both cases.

### ECharts option [ECharts configuration item] 
```
option = {
    xAxis: {
        type: 'category',
        axisTick: {
          alignWithLabel: true  
        },
        data: [
          'Campaign 1'
        , 'Campaign 2'
        , 'Campaign 3'
        , 'Campaign 4'
        , 'Campaign 5'
        , 'Campaign 6'
        , 'Campaign 7'
        , 'Campaign 8'
        , 'Campaign 9'
        , 'Campaign 10'
        ],
    },
    yAxis: {
        type: 'value'
    },
    series: [{
        data: [820, 932, 901, 934, 1290, 1330, 1320, 1320, 732, 920, ],
        type: 'line'
    }]
};

setTimeout(function () {
          
    myChart.setOption({
        xAxis: {
            axisLabel: {
                 rotate: '35'   
            }
        }
    });
    
});
```
### Other comments [Other Information]
Attaching the images for both cases:
**Case 1 : When window size is large**
![untitled](https://user-images.githubusercontent.com/24941920/50905133-19872980-1448-11e9-95dc-8eb5220489e0.jpg)

**Case 2 : When window size is small**
![untitled1](https://user-images.githubusercontent.com/24941920/50905142-1e4bdd80-1448-11e9-9d39-d98d912eb222.png)

**Comment by 100pah** [comment]
Does this example fit your scenario?
https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1

If so, this problem have been fixed in 4.1.0+ .

**Comment by 100pah** [comment]
Does this example fit your scenario?
https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1

If so, this problem have been fixed in 4.1.0+ .

**Comment by xerxes3117** [comment]
> Does this example fit your scenario?
> https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1
> 
> If so, this problem has been fixed in 4.1.0+ .

Thanks for the reply. 
I tried testing my code in the editor link that you provided ([https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1](url)). 
However, the issue still persists. Try adding my option code (in my question) in the editor, you'll notice that labels are still hidden (something similar to case 1 picture). However, if you reduce the width of the charts container it starts working at a certain width.

**Comment by xerxes3117** [comment]
> Does this example fit your scenario?
> https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1
> 
> If so, this problem have been fixed in 4.1.0+ .

Also, try doing this in your example ([https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1](url)). 
Reduce the number of data points in categoryData to `20` : 
```
for (var i = 0; i < 20; i++) {
    categoryData.push(i + 'longlonglong');
    data.push(Math.random() * 1000000);
}
```
You'll notice that x axis labels in top chart don't unhide, no matter what  the angle of rotation.

**Comment by xerxes3117** [comment]
> Does this example fit your scenario?
> https://gallery.echartsjs.com/editor.html?c=xbDq1UoC5j&v=1
> 
> If so, this problem have been fixed in 4.1.0+ .

Hi, is there any update on this issue?
